File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.py
class Solution:
    def numSub(self, s: str) -> int:
        res = 0
        n = 0
        for ch in s:
            if ch == '1':
                n += 1
            else:
                res += ((n * (n + 1) / 2) % 1000000007)
                n = 0
        res += ((n * (n + 1) / 2) % 1000000007)
        return int(res)

File: 1636-number-of-substrings-with-only-1s/1636-number-of-substrings-with-only-1s.cpp
class Solution {
public:
    int numSub(string s) {
        int res = 0;
        for(int i = 0; i < s.length(); i++)
        {
            long long n = 0;
            while(i < s.length() && s[i] == '1')
            {
                n++;
                i++;
            }
            res += ((n * (n + 1) / 2) % 1000000007);
        }
        return res;
    }
};